src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -2024,7 +1978,11 @@ else if (logger.isTraceEnabled())
     {
         if (!skipFlush)
         {
-            forceBlockingFlush();
+            Memtable current = getTracker().getView().getCurrentMemtable();
+            if (current.shouldSwitch(FlushReason.SNAPSHOT))
+                FBUtilities.waitOnFuture(switchMemtableIfCurrent(current, FlushReason.SNAPSHOT));
+            else
+                current.performSnapshot(snapshotName);
```

Comment:
is it OK that snapshot is created "twice" - once is `current.performSnapshot`, and second in `snapshotWithoutFlush`?
or is `performSnapshot` method name misleading?

Reply:
Renamed `snapshotWithoutFlush` to `snapshotWithoutMemtable` and `skipFlush` to `skipMemtable` to avoid the confusion.